ipa-core/src/net/client/mod.rs

```diff
@@ -153,39 +155,15 @@ impl Future for ResponseFuture {
 /// TODO: It probably isn't necessary to always use `[MpcHelperClient; 3]`. Instead, a single
 ///       client can be configured to talk to all three helpers.
 #[derive(Debug, Clone)]
-pub struct MpcHelperClient {
+pub struct MpcHelperClient<R: TransportRestriction = Ring> {
     client: Client<HttpsConnector<HttpConnector>, Body>,
     scheme: uri::Scheme,
     authority: uri::Authority,
     auth_header: Option<(HeaderName, HeaderValue)>,
+    _restriction: PhantomData<R>,
 }
 
-impl MpcHelperClient {
-    /// Create a set of clients for the MPC helpers in the supplied helper network configuration.
-    ///
-    /// This function returns a set of three clients, which may be used to talk to each of the
-    /// helpers.
-    ///
-    /// `identity` configures whether and how the client will authenticate to the server. It is for
-    /// the helper making the calls, so the same one is used for all three of the clients.
-    /// Authentication is not required when calling the report collector APIs.
-    #[must_use]
-    #[allow(clippy::missing_panics_doc)]
-    pub fn from_conf(
-        runtime: &IpaRuntime,
-        conf: &NetworkConfig,
-        identity: &ClientIdentity,
-    ) -> [MpcHelperClient; 3] {
-        conf.peers().each_ref().map(|peer_conf| {
-            Self::new(
-                runtime.clone(),
-                &conf.client,
-                peer_conf.clone(),
-                identity.clone_with_key(),
-            )
-        })
-    }
-
+impl<R: TransportRestriction> MpcHelperClient<R> {
```

Comment:
It is no longer an MPC client I think. It is fine to follow up on renaming, but lets leave a trace for this action item somewhere

Reply:
Why do you think it's not an "MPC" client? It's still used to talk to multiple parties in a protocol.

Or is it the "Helper" part that bothers you? I can just omit it and call it `MpcClient`?

Reply:
It is actually `Mpc` that bothers me. Shard connections shouldn't be confused with Mpc and having a struct that  has `Mpc` in its name but serving shard-to-shard connections looks confusing enough to me

Reply:
Mmm... What about `IpaClient<Helper>` and `IpaClient<Shard>` ?

Reply:
sure that looks great